Rubberduck.Main/Root/RubberduckIoCInstaller.cs

```diff
@@ -146,13 +147,41 @@ public void Install(IWindsorContainer container, IConfigurationStore store)
             RegisterParseTreeInspections(container, assembliesToRegister);
             RegisterInspections(container, assembliesToRegister);
             RegisterQuickFixes(container, assembliesToRegister);
+            RegisterCodeMetrics(container, assembliesToRegister);
 
             RegisterSpecialFactories(container);
             RegisterFactories(container, assembliesToRegister);
 
             ApplyDefaultInterfaceConvention(container, assembliesToRegister);
         }
 
+        private void RegisterCodeMetrics(IWindsorContainer container, Assembly[] assembliesToRegister)
+        {
+            var metricImplementations = new List<CodeMetric>();
+            foreach (var assembly in assembliesToRegister)
+            {
+                metricImplementations.AddRange(assembly.GetTypes()
+                    .Where(t => t.BaseType == typeof(CodeMetric))
+                    .Select(t =>
+                    {
+                        try
+                        {
+                            return (CodeMetric)Activator.CreateInstance(t, new object[] { });
```

Comment:
I kinda want a cleaner way than instantiating these at registration time

Reply:
Why do you not just register all the code metrics with `CodeMetric`? Then the collection of all of them will be registered with `IEnumerable<CodeMetric>`.

Reply:
I tried that before, it didn't work :/